src/FakeItEasy/Creation/CreationResult.cs

```diff
@@ -231,5 +243,22 @@ public int GetHashCode(ResolvedConstructor obj)
                 }
             }
         }
+
+        private sealed class CreationMode
+        {
+            private CreationMode(string name, Func<string, Exception> exceptionFactory)
+            {
+                this.Name = name;
+                this.ExceptionFactory = exceptionFactory;
+            }
+
+            public static CreationMode Fake { get; } = new CreationMode("fake", message => new FakeCreationException(message));
+
+            public static CreationMode Dummy { get; } = new CreationMode("dummy", message => new DummyCreationException(message));
+
+            public string Name { get; }
+
+            public Func<string, Exception> ExceptionFactory { get; }

```

Comment:
nitpicking: maybe expose this as a method rather than a delegate?

Reply:
Oh, yes. How foolish of me!

Reply:
Wait. A method that calls the delegate, or do you prefer me to subclass in order to have a method all the way down?

Reply:
Taking the easy route for now. Pushed.

Reply:
> A method that calls the delegate

That's what I had in mind